db/db_impl/db_impl_compaction_flush.cc

```diff
@@ -1965,11 +1971,21 @@ Status DBImpl::EnableAutoCompaction(
 }
 
 void DBImpl::DisableManualCompaction() {
-  manual_compaction_paused_.store(true, std::memory_order_release);
+  InstrumentedMutexLock l(&mutex_);
+  manual_compaction_paused_.fetch_add(1, std::memory_order_release);
+  // Wait for any pending manual compactions to finish (typically through
+  // failing with `Status::Incomplete`) prior to returning. This way we are
+  // guaranteed no pending manual compaction will commit while manual
+  // compactions are "disabled".
+  while (HasPendingManualCompaction()) {
+    bg_cv_.Wait();
+  }
 }
 
 void DBImpl::EnableManualCompaction() {
-  manual_compaction_paused_.store(false, std::memory_order_release);
+  InstrumentedMutexLock l(&mutex_);
+  assert(manual_compaction_paused_ > 0);
+  manual_compaction_paused_.fetch_add(-1, std::memory_order_release);
```

Comment:
Do we need a signal here?

Reply:
Any pending manual compactions were already finished (typically by failing with `Status::Incomplete`) during `DisableManualCompaction()`, so it's my understanding that nothing is waiting on this condition. `manual_compaction_paused_` is quite a confusing name as nothing seems to be paused.. 

Tangential - another name that confused me a lot in this PR is "incomplete", as a manual compaction returning `Status::Incomplete`, and a manual compaction setting `ManualCompactionState::incomplete`, mean completely different things :p.